### PR TITLE
Refactor references to willTransition (#1731)

### LINKED ISSUE #1731: Ember 4 prep - refactor references to willTransition and didTransition
willTransition and didTransition are removed in favor of router service events in Ember 4. [Guide here](https://deprecations.emberjs.com/v3.x/#toc_deprecate-router-events). We should start showing the router events instead.

We are using them on this page: https://guides.emberjs.com/release/routing/preventing-and-retrying-transitions

This PR can be opened and merged as soon as its ready. We don't have to wait for Ember 4, but we do need it in place by then.

### PR DESCRIPTION
**Pull request by rajakvk**
Closes #1731 
Remove references to deprecated `willTransition` and add `routeWillChange`
@jenweber Please review

### CODE REVIEW

**Comment by bertdeblock** [comment]
👋 Replacing `willTransition` with `routeWillChange` here isn't correct as Routes do not have the `routeWillChange` event. `routeWillChange` is [an event on the RouterService](https://api.emberjs.com/ember/3.27/classes/RouterService/events/routeWillChange?anchor=routeWillChange).

Maybe it's a good idea to reuse that example here as well?
- It uses `constructor` instead of `init`
- It doesn't access the controller (accessing the controller outside of the `setupController` and `resetController` hooks is [linted against](https://github.com/ember-cli/eslint-plugin-ember/blob/master/docs/rules/no-controller-access-in-routes.md))

The `action` import can be omitted though, it isn't used. I made a PR [here](https://github.com/emberjs/ember.js/pull/19743) to address this.

I think some of the copy around the code examples needs updating as well. For instance:
> When a transition is attempted, whether via `<LinkTo />`, `transitionTo`, or a URL change, a `willTransition` action is fired on the currently active routes.

Could be updated to something like:
> When a transition is attempted, whether via [`<LinkTo />`](https://api.emberjs.com/ember/release/classes/Ember.Templates.components/methods/LinkTo?anchor=LinkTo), [`transitionTo`](https://api.emberjs.com/ember/3.27/classes/RouterService/methods/transitionTo?anchor=transitionTo), or a URL change, a [`routeWillChange`](https://api.emberjs.com/ember/3.27/classes/RouterService/events/routeWillChange?anchor=routeWillChange) event is fired on the [`RouterService`](https://api.emberjs.com/ember/3.27/classes/RouterService).

Could be that some of the other copy needs a bit of updating as well.

**Comment by jenweber** [comment]
Hi, thanks for your PR! I will review it this week. If anyone else has time to review sooner, please go for it. I need a little time to test the example in an app.

**Review by jenweber — APPROVED** [review]
This is important prep for 4.0. Thank you!

File: guides/release/routing/preventing-and-retrying-transitions.md [hunk 1]
```diff
@@ -5,11 +5,10 @@ to immediately abort the transition by calling `transition.abort()`,
 and if the transition object is stored, it can be re-attempted at a
 later time by calling `transition.retry()`.
 
-### Preventing Transitions via `willTransition`
+### Preventing Transitions via `routeWillChange`
 
 When a transition is attempted, whether via `<LinkTo />`, `transitionTo`,
-or a URL change, a `willTransition` action is fired on the currently
-active routes. This gives each active route, starting with the leaf-most
+or a URL change, a `routeWillChange` event is fired on the [RouterService](https://api.emberjs.com/ember/release/classes/RouterService/events). This gives each active route, starting with the leaf-most
```
Comment (jenweber) [inline guides/release/routing/preventing-and-retrying-transitions.md:11]:
```suggestion
or a URL change, a `routeWillChange` event is fired on the [`RouterService`](https://api.emberjs.com/ember/release/classes/RouterService/events). This gives each active route, starting with the leaf-most
```
This change is to make the linter happy.

**Comment by mansona** [comment]
Percy is failing for some strange reason 🙈 I know what the issue is: the problem is that we have changed the browser since our last build on master 🙃 

After we merge this the master will build again, and future PRs will be happy 🎉

**Pull request merged**